### PR TITLE
[root-config] Fix --git-revision:

### PR DESCRIPTION
**Pull request by Axel-Naumann**
Fixes https://github.com/root-project/root/issues/14229

# This Pull request:

## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #

### CODE REVIEW

**Comment by pcanal** [comment]
Do we have a test for this?

**Comment by ferdymercury** [comment]
A test would be a good idea, yeah.

File: config/root-config.in [hunk 1]
```diff
@@ -558,7 +558,7 @@ while test $# -gt 0; do
     --git-revision)
       ### Output the git revision. Give up if gitinfo.txt can not be found.
       if test -r "${etcdir}/gitinfo.txt"; then
-         out="$out `head -m2 \"${etcdir}/gitinfo.txt\" | tail -n`"
+         out="$out `head -n 2 \"${etcdir}/gitinfo.txt\" | tail -n 1`"
```
Comment (ferdymercury) [inline config/root-config.in:561]: I think the space between n and the numbers is not needed, but it does not harm either.

Comment (ferdymercury) [reply config/root-config.in:561]: Other than that, I think there is still a problem. The etc/gitinfo.txt file for Ubuntu 22.04 binary release says:

```
tags/v6-30-02

Nov 27 2023, 19:50:38
```

so `head -n2 etc/gitinfo.txt | tail -n1`

returns an empty string.

**Review by ferdymercury — CHANGES_REQUESTED** [review]
sth is still weird with the gitinfo.txt on Ubuntu22 binary release

File: config/root-config.in (hunk 1 above)
Comment (ferdymercury) [reply config/root-config.in:561]: So it clearly does not match with the content of etc/gitinfo.txt that @bellenot showed in his post for Windows.

**Review by ferdymercury — APPROVED** [review]

**Comment by guitargeek** [comment]
@phsft-bot build

**Comment by guitargeek** [comment]
Thanks for the fix and the review!

**Comment by ferdymercury** [comment]
Thanks!
Is there a GitHub issue somewhere where we could collect 'future tests' ideas, so that they do not get lost inside issues?

Like:

- https://github.com/root-project/root/pull/14231#issuecomment-1855806343 
- https://github.com/root-project/root/pull/14275#issuecomment-1864499314
- https://github.com/root-project/root/pull/13217
- https://github.com/root-project/root/issues/14085

**Comment by guitargeek** [comment]
No, we don't have such an issue, but I think it's a great idea to have it. We can mark it also with the "good for new contributors" label, because writing tests is always a good way to get familiar with the codebase. Can you open such an issue, please? :pray:

**Pull request merged**